share/getters/cascade_test.go

```diff
@@ -53,8 +54,7 @@ func TestCascade(t *testing.T) {
 	successGetter := mocks.NewMockGetter(ctrl)
 	timeoutGetter.EXPECT().GetEDS(gomock.Any(), gomock.Any()).
 		DoAndReturn(func(ctx context.Context, _ *share.Root) (*rsmt2d.ExtendedDataSquare, error) {
-			<-ctx.Done()
-			return nil, ctx.Err()
+			return nil, context.DeadlineExceeded
```

Comment:
No idea 😬

Reply:
We should not merge this until we found the reason why this test is failing. as that could be a bug

Reply:
The thing is, i'm not sure how this test worked in the first place bc the context we pass in times out too late and we're waiting for the signal from Done.

Reply:
error observed starting here https://github.com/celestiaorg/celestia-node/commit/26e423d1332d82bbba13558a1624200f45715671

Reply:
Looks like my answer was not sent for some reason, so writing once again.

Commit that cleaned timeouts broke it. Test is waiting ctx timeout to happen before test returning (read from channel). It was the intermidate timeout that happen before that allowed the the test to pass. There is actually no need for this timeout in test. Intermidiate timeout was removed, because it was interfering with timeout passed by the caller. 
Fix looks good to me.

Reply:
Makes sense approing